children/schema.py

```diff
@@ -128,18 +128,54 @@ def resolve_past_enrolment_count(self: Child, info, **kwargs):
 
     def resolve_past_events(self: Child, info, **kwargs):
         """
-        Past events include Events the user has enrolled AND the occurrence of the
-        enrolment is more than KUKKUU_ENROLLED_OCCURRENCE_IN_PAST_LEEWAY mins in the
-        past.
+        Past events include:
+          * internal ticket system Events the user has enrolled and the end time of the
+            Enrolment's Occurrence is in the past
+          * external ticket system Events the user has a password to and the Event's
+            ticket_system_end_time is in the past
+
+        To be considered as being in the past an end time needs to be actually
+        KUKKUU_ENROLLED_OCCURRENCE_IN_PAST_LEEWAY mins in the past.
         """
-        events = self.project.events.user_can_view(info.context.user).published()
+        from events.schema import OccurrenceNode, TicketmasterEnrolmentNode  # noqa
+
+        published_events = self.project.events.user_can_view(
+            info.context.user
+        ).published()
 
         occurrences = self.occurrences.with_end_time()
+        now = timezone.now()
+        past_enough = now - timedelta(
+            minutes=settings.KUKKUU_ENROLLED_OCCURRENCE_IN_PAST_LEEWAY
+        )
         past_enough_enrolled_occurrences = occurrences.filter(
-            end_time__lt=timezone.now()
-            - timedelta(minutes=settings.KUKKUU_ENROLLED_OCCURRENCE_IN_PAST_LEEWAY)
+            end_time__lt=past_enough,
+            event__in=published_events,
+        ).select_related("event")
+
+        past_ticket_system_passwords = (
+            self.ticket_system_passwords.filter(
+                (
+                    ~Q(event__ticket_system=Event.INTERNAL)
+                    & Q(event__ticket_system_end_time__lt=now)
```

Comment:
now → past_enough?

Reply:
That is a good question, and I actually started implementing it like that first. However, there are multiple reasons why I decided to go with now instead. Explaining the whole thing would not be easy without a wall of text, so lets rather have an actual discussion about that elsewhere if needed. However, doing that either way should not really matter at all in practice, so it is questionable how much time we want to spend here 😄

Reply:
We can have a talk about it. IMHO the docstring should reflect what's implemented or at least it shouldn't be misleading i.e. give false information. Now the docstring says "To be considered as being in the past an end time needs to be actually KUKKUU_ENROLLED_OCCURRENCE_IN_PAST_LEEWAY mins in the past." and ..."Event's ticket_system_end_time is in the past". I read those together to mean that past_enough would be correct to be used here. I'm not saying you must use past_enough here though but if you don't use past_enough here then the docstring gives false or at least IMHO misleading information. So please would you either update the docstring to reflect the implementation or the implementation to reflect the docstring, thank you!

Reply:
I totally agree. I actually forgot to update the docstring when I changed the implementation 😄 Good catch! 👍 Fixed now.